ui/app/routes/playground/DatapointPlaygroundOutput.tsx

```diff
@@ -8,28 +8,206 @@ import {
+
     const query = useQuery({
       queryKey: getClientInferenceQueryKey(props),
       queryFn: getClientInferenceQueryFunction(props),
       // Only re-fetch when the user explicitly requests it
       refetchOnMount: false,
+      refetchOnWindowFocus: false,
       refetchInterval: false,
       retry: false,
     });
 
+    // For edited variants, track config changes to re-run evaluations
+    const variantConfig = variant.type === "edited" ? variant.config : null;
+
+    // Run evaluation when inference is ready and evaluation is selected
+    useEffect(() => {
+      if (!selectedEvaluation || !query.isSuccess || query.isRefetching) {
+        if (!selectedEvaluation) {
+          setEvalState({ status: "idle" });
+        }
+        return;
+      }
+
+      // Abort any previous request
+      abortControllerRef.current?.abort();
+      abortControllerRef.current = new AbortController();
+
+      setEvalState({ status: "loading" });
+
+      const runEval = async () => {
+        try {
+          // For builtin variants, pass variantName; for edited variants, pass variantConfig
+          const variantPayload =
+            variant.type === "builtin" || !variantConfig
+              ? { variantName: variant.name }
+              : {
+                  variantConfig: JSON.stringify(
+                    variantInfoToUninitializedVariantInfo(variantConfig),
+                  ),
+                };
+
+          const response = await fetch("/api/playground/evaluate", {
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({
+              evaluationName: selectedEvaluation,
+              ...variantPayload,
+              datapointIds: [datapoint.id],
+            }),
+            signal: abortControllerRef.current?.signal,
+          });
+
+          if (!response.ok) {
+            const errorText = await response.text();
+            setEvalState({ status: "error", error: errorText });
+            return;
+          }
+
+          const reader = response.body?.getReader();
+          if (!reader) {
+            setEvalState({ status: "error", error: "No response body" });
+            return;
+          }
+
+          const decoder = new TextDecoder();
+          let buffer = "";
+
+          const processLine = (line: string) => {
+            if (!line.trim()) return;
+            try {
+              const event = JSON.parse(line) as EvaluationEvent;
+              if (
+                event.type === "success" &&
+                event.datapoint?.id === datapoint.id
+              ) {
+                setEvalState({
+                  status: "success",
+                  result: {
+                    evaluations: event.evaluations ?? {},
+                    evaluatorErrors: event.evaluator_errors ?? {},
+                  },
+                });
+              } else if (
+                event.type === "error" &&
```

Comment:
can we sort the evaluations here? so they're displayed in a consistent order across datapoints?